src/main/java/org/folio/service/EventsGenerationService.java

```diff
@@ -56,32 +61,41 @@ private Future<SynchronizationJob> generateEventsRecursively(SynchronizationJob
       .onComplete(this::logEventsGenerationResult)
       .compose(page -> updateStats(job, page))
       .recover(error -> handleError(job, error))
-      .compose(syncJob -> fetchNextPage(syncJob, currentPage.get(), originalQuery));
+      .compose(syncJob -> fetchNextPage(syncJob, currentPage.get(), originalQuery))
+      .onSuccess(result -> log.info("generateEventsRecursively:: result: {}",
+        () -> logAsJson(result)));
   }
 
   private Future<List<T>> generateEventsForPage(List<T> page) {
+    log.debug("generateEventsForPage:: parameters page: list(size={})", page.size());
     return page.stream()
       .map(this::generateEvents)
       .reduce(succeededFuture(), (prev, next) -> prev.compose(r -> next))
-      .map(page);
+      .map(page)
+      .onSuccess(result -> log.info("generateEventsForPage:: result: list(size={})", result.size()));
   }
 
   private Future<SynchronizationJob> fetchNextPage(SynchronizationJob job, List<T> lastPage,
     String query) {
+    log.debug("fetchNextPage:: parameters job: {}, lastPage: list(size={}), query: {}",
+      () -> logAsJson(job), lastPage::size, () -> query);
 
     if (lastPage.size() < PAGE_SIZE) {
-      log.info("{} finished processing last page", getClass().getSimpleName());
+      log.info("fetchNextPage:: {} finished processing last page", getClass().getSimpleName());
       return succeededFuture(job);
     }
 
     T lastElement = lastPage.get(lastPage.size() - 1);
     String lastElementId = JsonObject.mapFrom(lastElement).getString("id");
     UuidHelper.validateUUID(lastElementId, true);
 
-    return generateEventsRecursively(job, query, lastElementId);
+    return generateEventsRecursively(job, query, lastElementId)
+      .onSuccess(result -> log.info("fetchNextPage:: result: {}", () -> logAsJson(job)));
   }
 
   private Future<SynchronizationJob> handleError(SynchronizationJob syncJob, Throwable error) {
+    String className = getClass().getSimpleName();
+    log.warn("{} failed to generate events", className, error);
```

Comment:
```suggestion
    log.warn("{} failed to generate events {}", className, error);
```

Reply:
error is Throwable, we don't need a placeholder for it in the template, stacktrace will be logged